### PR TITLE
Move email confirmation sending to a task for retries

### PR DESCRIPTION
**Pull request by robhudson**
We are getting many SMTP authentication errors, even when the auth is correct, so the task will retry these.

### CODE REVIEW

File: basket/petition/views.py [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
 from django.http import JsonResponse
 from django.shortcuts import redirect
-from django.template.loader import render_to_string
 from django.urls import reverse
 from django.utils.http import urlsafe_base64_decode, urlsafe_base64_encode
 from django.views.decorators.cache import cache_page
 
 from basket.petition.forms import PetitionForm
 from basket.petition.models import Petition
+from basket.petition.tasks import send_email_confirmation
```
Comment (pmac) [inline basket/petition/views.py:12]: I don't see this tasks file. Maybe it wasn't added to the commit by mistake?

**Review by stevejalim — APPROVED** [review]
LGTM

File: basket/petition/views.py [hunk 2]
```diff
@@ -59,7 +40,9 @@ def sign_petition(request):
         petition.save()
```
Comment (pmac) [inline basket/petition/views.py:40]: should we also catch a potential DB error here and specifically return an error, or just let it 500?

**Review by pmac — APPROVED** [review]
:shipit:

**Pull request merged**